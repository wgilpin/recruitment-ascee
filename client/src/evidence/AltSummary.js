import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  corporations: PropTypes.array, 
  selected: PropTypes.string, 
  corporationId: PropTypes.number, 
  corpRedlisted: PropTypes.bool,
  corporationName: PropTypes.string,
  secStatus: PropTypes.number,
  onClickCorp: PropTypes.func,
};

const defaultProps = {};

const styles = {
  corporation: {
    maxWidth: '300px',
  },
  corpCEO: {
    cursor: 'pointer',
    textDecoration: 'underline',
    color: '#01799A',
    maxWidth: '300px',
  },
  secStatus: {
    fontWeight: 500,
  },
  red: {
    color: 'red',
    fontWeight: 500,
  },
  summary: {
    backgroundColor: '#222',
    padding: '4px',
  },
};

export default function AltSummary(props) {
  const { corporations, selected, corporationId, corpRedlisted, corporationName, secStatus } = props;
  if (!corporations) {
    return;
  }
  const isCEO =
    (corporations[corporationId] || {}).ceo_id === parseInt(selected);
  let corpStyle = isCEO ? styles.corpCEO : styles.corporation;
  if (corpRedlisted) {
    corpStyle = { ...corpStyle, ...styles.red };
  }
  return (
    <div style={styles.summary}>
      <div>
        <div style={styles.secStatus}>
          Sec Status: {Math.round(secStatus * 100) / 100}
        </div>
        <div>
          Corp {isCEO && '[CEO]'}&emsp;
          <span
            style={corpStyle}
            onClick={() => props.onClickCorp(corporationId)}
          >
            {corporationName}
          </span>
        </div>
      </div>
    </div>
  );
}

AltSummary.propTypes = propTypes;
AltSummary.defaultProps = defaultProps;
