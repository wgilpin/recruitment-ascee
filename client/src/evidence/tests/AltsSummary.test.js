import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import AltSummary from '../AltSummary';

const corps =  [
  {
    ceo_id: 93908806,
    ceo_name: 'Midgely Urea',
    corporation_name: 'Winnits and Other Things',
  },
  {
    ceo_id: 2113387072,
    ceo_name: 'Kate Flashheart',
    corporation_name: "Kate's Manservant Recruitment Agency",
  },
]

const char_93908806 = {
  id: 93908806,
  corporation_id: 98503673,
  corporation_name: 'Winnits and Other Things',
  name: 'The Flight Lieutenant',
}

describe('<AltSummary>', () => {
  it('matches snapshot', async () => {
    const adminConfig = mount(
      <AltSummary
      corporations={corps}
      selected={char_93908806.id}
      corporationId={char_93908806.corporation_id}
      corpRedlisted={false}
      corporationName={char_93908806.corporation_name}
      secStatus={5.0}
      />
    );
    const tree = renderer.create(adminConfig);
    await Promise.resolve();
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
