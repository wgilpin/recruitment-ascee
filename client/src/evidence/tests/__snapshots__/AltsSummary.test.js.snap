// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AltSummary> matches snapshot 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#222",
      "padding": "4px",
    }
  }
>
  <div>
    <div
      style={
        Object {
          "fontWeight": 500,
        }
      }
    >
      Sec Status: 
      5
    </div>
    <div>
      Corp 
      â€ƒ
      <span
        onClick={[Function]}
        style={
          Object {
            "maxWidth": "300px",
          }
        }
      >
        Winnits and Other Things
      </span>
    </div>
  </div>
</div>
`;
